com.github.dockerjava.api.model.PullResponseItem#isPullSuccessIndicated fails to check for "Exists" status

com.github.dockerjava.api.model.PullResponseItem#isPullSuccessIndicated checks for several success states, but misses "Exists" and treats this status as a failure.

Caused by: com.github.dockerjava.api.exception.DockerClientException: Could not pull image: Exists

If would be great if the success status codes could be a configurable list that the library user could append to.

--- a/docker-java-api/src/main/java/com/github/dockerjava/api/model/PullResponseItem.java
+++ b/docker-java-api/src/main/java/com/github/dockerjava/api/model/PullResponseItem.java
@@ -19,6 +19,8 @@ public class PullResponseItem extends ResponseItem {
 
     private static final String DOWNLOADED_SWARM = ": downloaded";
 
+    private static final String ALREADY_EXISTS = "Already exists";
+
     /**
      * Returns whether the status indicates a successful pull operation
      *
@@ -34,7 +36,8 @@ public class PullResponseItem extends ResponseItem {
                 getStatus().contains(IMAGE_UP_TO_DATE) ||
                 getStatus().contains(DOWNLOADED_NEWER_IMAGE) ||
                 getStatus().contains(LEGACY_REGISTRY) ||
-                getStatus().contains(DOWNLOADED_SWARM)
+                getStatus().contains(DOWNLOADED_SWARM)||
+                getStatus().contains(ALREADY_EXISTS)
         );
     }
 }

--- a/docker-java-api/src/main/java/com/github/dockerjava/api/model/PullResponseItem.java
+++ b/docker-java-api/src/main/java/com/github/dockerjava/api/model/PullResponseItem.java
@@ -36,7 +36,7 @@ public class PullResponseItem extends ResponseItem {
                 getStatus().contains(IMAGE_UP_TO_DATE) ||
                 getStatus().contains(DOWNLOADED_NEWER_IMAGE) ||
                 getStatus().contains(LEGACY_REGISTRY) ||
-                getStatus().contains(DOWNLOADED_SWARM)||
+                getStatus().contains(DOWNLOADED_SWARM) ||
                 getStatus().contains(ALREADY_EXISTS)
         );
     }

--- a/docker-java/src/test/java/com/github/dockerjava/api/model/PullResponseItemTest.java
+++ b/docker-java/src/test/java/com/github/dockerjava/api/model/PullResponseItemTest.java
@@ -30,6 +30,14 @@ import static org.junit.Assert.assertTrue;
  */
 public class PullResponseItemTest {
     @Test
+    public void imageAlreadyExists() throws IOException {
+        PullResponseItem response = testRoundTrip(PullResponseJSONSamples.pullImageResponse_alreadyExists,
+                PullResponseItem.class);
+        assertTrue(response.isPullSuccessIndicated());
+        assertFalse(response.isErrorIndicated());
+    }
+
+    @Test
     public void pullNewerImage() throws IOException {
         PullResponseItem response = testRoundTrip(PullResponseJSONSamples.pullImageResponse_newerImage,
                 PullResponseItem.class);

--- a/docker-java/src/test/java/com/github/dockerjava/api/model/PullResponseJSONSamples.java
+++ b/docker-java/src/test/java/com/github/dockerjava/api/model/PullResponseJSONSamples.java
@@ -23,7 +23,9 @@ import com.github.dockerjava.test.serdes.JSONResourceRef;
  * @author Zach Marshall
  */
 public enum PullResponseJSONSamples implements JSONResourceRef {
-    pullImageResponse_legacy, pullImageResponse_error, pullImageResponse_newerImage, pullImageResponse_upToDate;
+    pullImageResponse_legacy, pullImageResponse_error, 
+    pullImageResponse_newerImage, pullImageResponse_upToDate,
+    pullImageResponse_alreadyExists;
 
     @Override
     public String getFileName() {

--- a/docker-java/src/test/resources/com/github/dockerjava/api/model/pullImageResponse_alreadyExists.json
+++ b/docker-java/src/test/resources/com/github/dockerjava/api/model/pullImageResponse_alreadyExists.json
@@ -1,0 +1,1 @@
+{"status":"Downloaded newer image for docker.com/xxxxcv/ccccxx"}

--- a/docker-java/src/test/resources/com/github/dockerjava/api/model/pullImageResponse_alreadyExists.json
+++ b/docker-java/src/test/resources/com/github/dockerjava/api/model/pullImageResponse_alreadyExists.json
@@ -1,1 +1,1 @@
-{"status":"Downloaded newer image for docker.com/xxxxcv/ccccxx"}
+{"status":"Already exists"}

